user/db.go

```diff
@@ -58,10 +58,17 @@ func (s *Store) updateUser(ctx context.Context, u User) error {
 	if err != nil {
 		panic(fmt.Errorf("failed to build sql for updateUser: %w", err))
 	}
-	_, err = s.db.ExecContext(ctx, sql, args...)
+	res, err := s.db.ExecContext(ctx, sql, args...)
 	if err != nil {
 		return fmt.Errorf("failed to update user: %w", err)
 	}
+	rows, err := res.RowsAffected()
+	if err != nil {
+		return fmt.Errorf("failed to update user: %w", err)
+	}
+	if rows < 1 {
+		return fmt.Errorf("failed to update user: invalid rows affected: %d", rows)
```

Comment:
Can we give a more meaningful error message - what does "0 rows affected" actually mean?

Reply:
Added a bit more context